src/main/java/github/scarsz/discordsrv/api/SlashCommandManager.java

```diff
@@ -0,0 +1,133 @@
+ * END
+ */
+
+package github.scarsz.discordsrv.api;
+
+import github.scarsz.discordsrv.DiscordSRV;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import net.dv8tion.jda.api.exceptions.ErrorResponseException;
+import net.dv8tion.jda.api.exceptions.RateLimitedException;
+import net.dv8tion.jda.api.interactions.commands.build.CommandData;
+import net.dv8tion.jda.api.requests.ErrorResponse;
+import net.dv8tion.jda.api.requests.restaction.CommandListUpdateAction;
+import org.bukkit.plugin.Plugin;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+@RequiredArgsConstructor
+public class SlashCommandManager {
+    @Getter
+    private boolean updated = false;
+    private final Map<Plugin, PluginCommands> commandsMap = new HashMap<>();
+
+    public void reloadSlashCommands() {
+        CommandListUpdateAction action = DiscordSRV.getPlugin().getMainGuild().updateCommands();
+        boolean added = false;
+        for (PluginCommands value : commandsMap.values()) {
+            if (value.plugin.isEnabled()) {
+                added = true;
+                action.addCommands(value.data);
+            }
+        }
+        if (!added) return; //no need to register if no commands will be registered
+        action.queue(s -> updated = true,er -> printSlashRegistrationError(RegistrationResult.getResult(er), commandsMap.keySet()));
+    }
+
+    /**
+     * Sets the slash commands for a specific plugin, overrides old ones. this method does not register them, see {@link SlashCommandManager#reloadSlashCommands()}
+     * @param plugin Plugin that is registering the commands, whenever discordsrv is registering the commands it will not register the commands for that plugin if it is disabled
+     * @param commandData Commands to add
+     */
+    public void setCommandsForPlugin(@NotNull Plugin plugin, @NotNull CommandData... commandData) {
+        PluginCommands pluginCommands = commandsMap.get(plugin);
+        if (pluginCommands == null) {
+            pluginCommands = new PluginCommands(plugin, new HashSet<>(Arrays.asList(commandData)));
+            commandsMap.put(plugin, pluginCommands);
+        } else pluginCommands.data = new HashSet<>(Arrays.asList(commandData));
+        updated = false;
+    }
+
+    private void printSlashRegistrationError(RegistrationResult result, Set<Plugin> plugins) {
+        DiscordSRV.getPlugin().getLogger().warning("==============================================================");
+        DiscordSRV.getPlugin().getLogger().warning("DiscordSRV could not register slash commands to your discord server!");
+        switch (result) {
+            case SUCCESS:
+                break;
+            case RATE_LIMIT:
+                DiscordSRV.getPlugin().getLogger().warning("Rate Limited! Are you restarting your server or reloading this plugin alot?");
+                break;
```

Comment:
I don't think such a big warning is required for being rate limited, 1 line should be enough